react/MenuItem.tsx

```diff
@@ -70,6 +71,11 @@ const MenuItem: StorefrontFunctionComponent<MenuItemSchema> = ({
     [isActive]
   )
 
+  // Close any active/open menu when url changes
+  useUrlChange(() => {
+    if (isActive) setActive(false)
+  }, [isActive, setActive])
```

Comment:
Maybe we should use a `React.useCallback` to this since is registring every time the passed function changes

```suggestion
  const MyFn = React.useCallback(() => {
    if (isActive) setActive(false)
  }, [isActive])
  // Close any active/open menu when url changes
  useUrlChange(MyFn, [MyFn])
```

Reply:
Not so sure about this. We would still be defining the anonymous function, but inside the `useCallback` argument. Check: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Yes, but that's not my point. My point is that you're defining an anonymous function in each render and passing it to the `useUrlChange` [which uses this function in the array dependency of an useEffect](https://github.com/vtex-apps/menu/pull/116/files#diff-12a70a522b9a9f8a9d2e283b204d192a69bcc78f14ef54399249ce4f9690daf9R12) which means that this `useEffect` will be executed in every render as well. In the end, we won't have a performance improvement, but I think that you should use an `useCallback` or just remove the array dependency of the `useEffect` inside `useUrlChange` because it always says that the `useEffect` should execute

Reply:
Good point

Reply:
Got it! Nice catch 😁